### PR TITLE
+

### PR DESCRIPTION
**Pull request by AndriZhok**

### CODE REVIEW

File: app/cafe.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from app.errors import NotVaccinatedError
+from app.errors import OutdatedVaccineError
+from app.errors import NotWearingMaskError
```
Comment (Dimosphen1) [inline app/cafe.py:3]: Replace this import with the following
```suggestion
from app.errors import (
    NotVaccinatedError,
    OutdatedVaccineError,
    NotWearingMaskError
)
```
There is no need to duplicate `from app.errors import`

File: app/cafe.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+import datetime
```
Comment (Dimosphen1) [inline app/cafe.py:6]: Order your imports correctly to follow PEP-8 import rules

File: app/cafe.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from app.errors import NotVaccinatedError
+from app.errors import OutdatedVaccineError
+from app.errors import NotWearingMaskError
+
```
Comment (Dimosphen1) [inline app/cafe.py:4]: There is no need to add two blank lines between different import groups, one is enough

File: app/cafe.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    def __init__(self, cafe_name: str) -> None:
+        self.cafe_name = cafe_name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if "vaccine" not in visitor:
+            raise NotVaccinatedError("Visitor is not vaccinated")
+
+        vaccine_info = visitor["vaccine"]
+        expiration_date = vaccine_info["expiration_date"]
+        if expiration_date < datetime.date.today():
```
Comment (Dimosphen1) [inline app/cafe.py:19]: There is no need to define multiple arguments, perform retrieval in the if condition instead

File: app/main.py [hunk 5]
```diff
@@ -1 +1,24 @@
-# write your code here
+from app.cafe import Cafe
+
```
Comment (Dimosphen1) [inline app/main.py:2]: There is no need to separate imports from the same import group with blank line

File: app/main.py [hunk 6]
```diff
@@ -1 +1,24 @@
...
+from app.errors import NotVaccinatedError
+from app.errors import OutdatedVaccineError
+from app.errors import NotWearingMaskError
```
Comment (Dimosphen1) [inline app/main.py:5]: Change the import according to previous recommendations

File: app/main.py [hunk 7]
```diff
@@ -1 +1,24 @@
...
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotVaccinatedError:
+            return "All friends should be vaccinated"
+        except OutdatedVaccineError:
+            return "All friends should be vaccinated"
```
Comment (Dimosphen1) [inline app/main.py:16]: Think about how you can avoid this code duplication. Pay attention to exception inheritance hierarchy

File: app/main.py [hunk 8]
```diff
@@ -1 +1,24 @@
...
+        except NotWearingMaskError as e:
+            if str(e) == "Visitor's not wearing a mask":
```
Comment (Dimosphen1) [inline app/main.py:18]: There is no need to check the string here, since it is the only option for `NotWearingMaskError `

File: app/main.py [hunk 9]
```diff
@@ -1 +1,24 @@
-# write your code here
+from app.cafe import Cafe
+
+from app.errors import NotVaccinatedError
+from app.errors import OutdatedVaccineError
+from app.errors import NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
```
Comment (Dimosphen1) [inline app/main.py:8]: Indicate a list of what instances you expect to obtain for `friends` in the type annotation

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

**Review by pavlopro — CHANGES_REQUESTED** [review]
Please take into consideration all of previous comments

**Pull request closed without merging**